numpy/lib/index_tricks.py

```diff
@@ -769,9 +769,9 @@ def fill_diagonal(a, val, wrap=False):
     a : array, at least 2-D.
       Array whose diagonal is to be filled, it gets modified in-place.
 
-    val : scalar
-      Value to be written on the diagonal, its type must be compatible with
-      that of the array a.
+    val : scalar or array_like
+      Value or values to write on the diagonal; type must be compatible with that of array a. If an array of values, the length of values must be equal to the 
+      main diagonal to fill.
```

Comment:
Are you sure about this second sentence?

Reply:
I wrote the second line after trying out `a = np.zeros((4, 5)); np.fill_diagonal(a, [1, 2])` which returned `None`. I could edit the second line again based on what would be more accurate

Reply:
Doesn't it always return `None`?

Reply:
Ohhh good point, sorry I got confused. :'(. After running it just now I can see that it modifies the array in place and fills the values no matter the length. How would you like for me to word the case of passing in an array?